src/braket/circuits/unitary_calculation.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numpy as np
+
+from braket.circuits.circuit import Circuit
+from braket.circuits.gate import Gate
+from braket.circuits.qubit_set import QubitSet
+
+
+def _einsum_subscripts(targets: QubitSet, qubit_count: int) -> str:
+    target_count = len(targets)
+
+    gate_left_indexes = list(range(target_count))
+    un_left_indexes = list(range(target_count, target_count + qubit_count))
+    un_right_indexes = list(range(target_count + qubit_count, target_count + 2 * qubit_count))
+
+    gate_right_indexes = [un_left_indexes[-1 - target] for target in targets]
+
+    result_left_indexes = un_left_indexes.copy()
+    for pos, target in enumerate(targets):
+        result_left_indexes[-1 - target] = gate_left_indexes[pos]
+
+    return (
+        gate_left_indexes + gate_right_indexes,
+        un_left_indexes + un_right_indexes,
+        result_left_indexes + un_right_indexes,
+    )
+
+
+def as_unitary(circuit: Circuit) -> np.ndarray:
```

Comment:
Apologies if I wasn't clear earlier. We still want `as_unitary()` to be a method on the Circuit class. The implementation  for that method resides in this file. This is to try and not bloat the circuit.py file, in order to help with the maintainability.

```
from braket.circuits.unitary_calculation import calculate_unitary


class Circuit:

   def as_unitary(self):
       """
       <doc strings as before>
       """
       <validation (if any) >
        calculate_unitary(...)
      
```

Reply:
Ouch, my bad :(

Hopefully this time, it looks better: done in https://github.com/aws/amazon-braket-sdk-python/pull/253/commits/f14f8a9d0f25867d549faecde8d6dee75cb740d3